tests/testthat/test-checkstyle_output.R

```diff
@@ -33,5 +33,5 @@ test_that("return lint report as checkstyle xml", {
 
   # The second line is the checkstyle version, so we ignore it during the
   # check, so we don't have to update the version every release.
-  expect_equal(readLines(tmp)[-2L], readLines(test_path("checkstyle.xml"))[-2L])
```

Comment:
I have memorized that `expect_equal()` is preferable in testthat 3e because it has a better failure message. Am I right or is my memory flawed?

Reply:
IIUC the engine is the same. `expect_equal()` might be considered preferable for the latter case here?

```r
library(testthat)
Sys.setenv(TESTTHAT_EDITION = 3)
expect_identical(1:10, c(1:9, 11L))
# Error: 1:10 (`actual`) not identical to c(1:9, 11L) (`expected`).

#   `actual[7:10]`: 7 8 9 10
# `expected[7:10]`: 7 8 9 11
expect_equal(1:10, c(1:9, 11L))
# Error: 1:10 (`actual`) not equal to c(1:9, 11L) (`expected`).

#   `actual[7:10]`: 7 8 9 10
# `expected[7:10]`: 7 8 9 11

expect_identical(1:10, c(1:9, 11))
# Error: 1:10 (`actual`) not identical to c(1:9, 11) (`expected`).

# `actual` is an integer vector (1, 2, 3, 4, 5, ...)
# `expected` is a double vector (1, 2, 3, 4, 5, ...)
expect_equal(1:10, c(1:9, 11))
# Error: 1:10 (`actual`) not equal to c(1:9, 11) (`expected`).

#   `actual[7:10]`: 7 8 9 10
# `expected[7:10]`: 7 8 9 11
```

Personally I prefer being stricter in tests... moreover it's a consistency thing at this point for our suite (these are the only two `expect_equal()` left, vs. ~300 `expect_identical()`).

Reply:
Okay then that used to be.
I saw a failing test in the random order test suite @IndrajeetPatil attempted, reading `settings$encoding` is not identical to `"UTF-8"`, but not showing the actual value. But maybe that was just truncated.